tools/nl/common_name_entity/main.py

```diff
@@ -0,0 +1,154 @@
+    'input_gcs_bucket',
+    '',
+    'GCS bucket to download input files from. Will not download from gcs if this is not set.',
+    short_name='g')
+
+
+# Downloads files from a gcs bucket to input folder
+def get_files_from_gcs(gcs_bucket):
+  sc = storage.Client()
+  bucket = sc.get_bucket(gcs_bucket)
+  blobs = bucket.list_blobs()
+  for b in blobs:
+    b.download_to_filename(f'{_INPUT_FOLDER}{b.name}')
+
+
+# Returns a map of entity name to entity type where the entity name is flagged.
+# An entity name will be flagged if all its tokens are flagged and there is no
+# "-" or ":" within the name.
+def _generate_flagged_names(flagged_tokens, name2tokens, name2types):
+  flagged_names = {}
+  for n, tokens in name2tokens.items():
+    if "-" in n or ":" in n:
+      continue
+    if any([flagged_tokens.get(t, '') == '' for t in tokens]):
+      continue
+    flagged_names[n] = name2types.get(n, '')
+  return flagged_names
+
+
+# Gets a map of name of entity to type of entity
+def _get_name_to_type(file_df):
+  name_to_type = {}
+  for _, row in file_df.iterrows():
+    row_name = str(row['name'])
+    name_to_type[row_name] = row['type']
+  return name_to_type
+
+
+# Takes a list of names and chunks it into lists of names where the total number
+# of characters in each chunk (when each name is separated by a space) is less
+# than the max characters spacy can take
+def _get_chunked_names(names):
+  chunked_names = []
+  curr_chunk_names = []
+  curr_char_length = 0
+  for name in names:
+    if len(name) + curr_char_length + 1 > _MAX_SPACY_CHARS:
+      chunked_names.append(curr_chunk_names)
+      curr_char_length = 0
+    curr_char_length += len(name) + 1
+    curr_chunk_names.append(name)
+  chunked_names.append(curr_chunk_names)
+  return chunked_names
+
+
+# Gets the flagged tokens and flagged names for a chunk of names
+# flagged_tokens is a map of token to its part of sentence for tokens that are
+# flagged as possibly being a common word
+# flagged_names is a map of name to its entity type for names where all its
+# tokens are flagged
+def _get_results_for_names(nlp, names, name_to_type):
+  if not names:
+    return {}, {}
+  name_str = ' '.join(names)
+  doc = nlp(name_str)
+  # map of name to its list of tokens that make up the name
+  name_to_tokens = {}
+  # index in names list of the current name we are processing
+  curr_name_idx = 0
+  # the part of the current name that has yet to be processed
+  running_curr_name = names[curr_name_idx]
+  flagged_tokens = {}
+  for token in doc:
+    # If running_curr_name is empty, we've finished getting the tokens for
+    # this name, so move on to the next name
+    if not running_curr_name and curr_name_idx < len(names) - 1:
+      curr_name_idx += 1
+      running_curr_name = names[curr_name_idx]
+    # If running_curr_name starts with the token text, that means this token
+    # applies to this name. Add the token to name_to_tokens and update
+    # running_curr_name
```

Comment:
I think an example might be helpful here?

For instance its not clear why you're testing startswith instead of equality.

Reply:
added more comments